src/util/asmap.cpp

```diff
@@ -21,140 +21,235 @@
 #include <utility>
 #include <vector>
 
+/*
+ * ASMap (Autonomous System Map) Implementation
```

Comment:
All the documentation changes in this commit look correct to me.

One thing that may be worth mentioning early on is that it's a bit-packed format, i.e., the entire compressed mapping is treated as a sequence of bits, packed together at 8 per byte, which are concatenated encodings of instructions.